app/logic/Mail/ActiveSync/ActiveSyncAccount.ts

```diff
@@ -245,6 +263,10 @@ export class ActiveSyncAccount extends MailAccount {
       if (!wbxmljs.Status || wbxmljs.Status == "1" || aCommand == "Ping") {
         return wbxmljs;
       }
+      if (wbxmljs.Status == "111") {
```

Comment:
This is "Retryable server error", right?
1. Did you see exactly this error, in reality, when we're running into throttling on the server? Are there any other error calls that might appear in this case?
2. In case this status 111 happens for reasons other than throttling, and the server keeps throwing this error for 24 hours. What would prevent us from infinitely re-trying? We need some sort of counter that limits the number of times we try. You could express that as `RetryCounter` class as well.
3. We need to avoid that other calls keep hitting the server at the same time, and run into the same error as well.

The following would address 3, but it's still missing the `RetryCounter`:
```
      if (wbxmljs.Status == "111") { // Retryable server error
        let lock = await this.semaphore.lock();
        await sleep(5);
        lock.release();
        return await this.callEAS(aCommand, aRequest, heartbeat);
      }
```

Reply:
Or better yet, the following would address 3, but it's still missing the `RetryCounter`:
```
      if (wbxmljs.Status == "111") { // Retryable server error
        await this.throttle.waitFor(5);
        return await this.callEAS(aCommand, aRequest, heartbeat);
      }
```

Reply:
Implemented in git commit 874d67e1

Reply:
> 1. Did you see exactly this error, in reality, when we're running into throttling on the server?

No, I only added the logic because you'd said you'd seen that error. (I didn't think to click on the all folders account myself even though I know that generates a silly amount of parallel requests.)